lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -1083,7 +1083,7 @@ static LogicalResult buildAllocOp(ComponentLoweringState &componentState,
   SmallVector<int64_t> sizes;
   for (int64_t dim : memtype.getShape()) {
     sizes.push_back(dim);
-    addrSizes.push_back(llvm::Log2_64_Ceil(dim));
+    addrSizes.push_back(llvm::Log2_64_Ceil(dim) | 1);
```

Comment:
Can this be placed in a helper function instead of duplicated twice? Perhaps linking the issue you opened, since it mentions why (currently) Calyx doesn't support zero width.

Reply:
maybe but seems like abstraction for the sake of abstraction. better would be (on @mikeurbach's advice) the more recognziable form of `max(..., 1)`. i have to read that issue more closely before i decide whether this patch is a good idea at all.

Reply:
I agree that std::max is better too. I still think it should be placed in a helper function.